tpot/base.py

```diff
@@ -298,19 +299,35 @@ def _setup_config(self, config_dict):
                         'TPOTClassifier instead.'
                     )
             else:
-                try:
-                    with open(config_dict, 'r') as input_file:
-                        file_string = input_file.read()
-                    self.config_dict = eval(file_string[file_string.find('{'):(file_string.rfind('}') + 1)])
-                except Exception:
-                    raise TypeError(
-                        'The operator configuration file is in a bad format or '
-                        'not available. Please check the configuration file '
-                        'before running TPOT.'
-                    )
+                self._read_config_file(config_dict)
         else:
             self.config_dict = self.default_config_dict
 
+    def _read_config_file(self, config_path):
+        try:
+            custom_config = imp.new_module('custom_config')
+
+            with open(config_path, 'r') as config_file:
+                file_string = config_file.read()
+                exec(file_string, custom_config.__dict__)
+
+            self.config_dict = custom_config.tpot_config
```

Comment:
Very nice function. I think maybe returning the `custom_config.tpot_config` and assign `self.config_dict` in `_setup_config()` above is a little bit more consistent in structure.